src/mem/x86_64.rs

```diff
@@ -0,0 +1,91 @@
+use super::c_int;
+
+// On recent Intel processors, "rep movsb" and "rep stosb" have been enhanced to
+// select the best microarchitecural implemenetation based on length and
+// alignment. This allows for very fast memcmp/memset implementations.
+// See "Intel® 64 and IA-32 Architectures Optimization Reference Manual"
+//  - Enhanced REP MOVSB and STOSB operation (ERMSB) (section 3.7.7)
+//  - Fast Short REP MOV (FSRM)
+//
+// While the "rep cmpsb" instructions have not seen similar optimizations, they
+// are still faster than our simple fallback implementations for memcmp/bcmp.
+
+#[inline(always)]
+unsafe fn copy_forward(dest: *mut u8, src: *const u8, count: usize) {
+    asm!(
```

Comment:
:sob: cg_clif doesn't support inline asm. Can you add a feature flag to disable `asm!` usage?

Reply:
This isn't the only ASM in this crate, as many of the functions in this crate must me implemented in assembly. How is cg_clif currently building this crate if it doesn't support inline ASM?

Or is this a distinction between `llvm_asm!` and `asm!`

Reply:
I think it's ok to not support cg_clif in the first pass, that's not an officially supported backend and it's not at the point where it should be adding maintenance burden to this crate. @bjorn3 you're welcome to have a follow-up PR to update this crate if it stops working.

Reply:
The other inline asm is only used for stack probes. For this reason I disabled stack probes in Cranelift and made cg_clif recognize the specific inline asm of the stack probes and emit a trap (for `___chkstk_ms`) or nothing (for the `__rust_probestack` `global_asm!`. Combined this means that no inline asm is ever used. This PR would however introduce mandatory inline asm, that I would then need to emulate.

Reply:
I gated the changes behind the `asm` feature flag.